### PR TITLE
feat: batch create with custom network sample

### PR DESCRIPTION
**Pull request by BigBlackWolf**
## Description

Samples, which demonstrates how to create batch with custom network and subnetwork. Documentation reference [here](https://cloud.google.com/batch/docs/specify-job-network#create-job-specific-network)

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: batch/create_with_custom_network_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	expectedNetwork := fmt.Sprintf("projects/%s/global/networks/%s", tc.ProjectID, networkName)
+	expectedSubnetwork := fmt.Sprintf("projects/%s/regions/%s/subnetworks/%s", tc.ProjectID, region, subnetworkName)
+
+	interfaces := job.GetAllocationPolicy().GetNetwork().GetNetworkInterfaces()
+	if interfaces[0].GetNetwork() != expectedNetwork || interfaces[0].GetSubnetwork() != expectedSubnetwork {
+		t.Errorf("Network wasn't set")
+	}
+
+	if err := deleteJob(buf, tc.ProjectID, region, jobName); err != nil {
```
[the remark this answers is not in the document]
Comment (BigBlackWolf) [reply batch/create_with_custom_network_test.go:55]: Makes sense, added above

File: batch/create_with_custom_network.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	batch "cloud.google.com/go/batch/apiv1"
+	"cloud.google.com/go/batch/apiv1/batchpb"
+	durationpb "google.golang.org/protobuf/types/known/durationpb"
+)
+
+// Creates and runs a job with custom network
```
Comment (telpirion) [inline batch/create_with_custom_network.go:28]: nit: typical Godoc style is to begin a function-level comment with the name of the function. I'm also fine with this as-is, too.

File: batch/create_with_custom_network.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("batchClient error: %w", err)
+	}
+	defer batchClient.Close()
+
+	runn := &batchpb.Runnable{
+		Executable: &batchpb.Runnable_Script_{
+			Script: &batchpb.Runnable_Script{
+				Command: &batchpb.Runnable_Script_Text{
+					Text: "echo Hello world from script 1 for task ${BATCH_TASK_INDEX}",
```
Comment (telpirion) [inline batch/create_with_custom_network.go:41]: question: does this print out a message to a terminal somewhere? If so, you might want to clarify with a comment above this `Command` field.

File: batch/create_with_custom_network_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestCreateJobWithCustomNetwork(t *testing.T) {
+	var r *rand.Rand = rand.New(
```
Comment (telpirion) [inline batch/create_with_custom_network_test.go:29]: nit: I think you can just use the `:=` operator here.

**Review by telpirion — APPROVED** [review]

File: batch/create_with_custom_network.go (hunk 2 above)
Comment (BigBlackWolf) [reply batch/create_with_custom_network.go:28]: Oh, my bad, typo. 
Thanks, for the comments! I will extend near draft PR with your suggestions https://github.com/GoogleCloudPlatform/golang-samples/pull/4879

File: batch/create_with_custom_network_test.go (hunk 4 above)
Comment (BigBlackWolf) [reply batch/create_with_custom_network_test.go:29]: I stuck to this approach to follow the pattern in other tests. I don't have a strong preference though. Perhaps, it is even better and will allow to get rid of the line break. I will adjust.

**Pull request merged**